Source/Core/Resource.js

```diff
@@ -374,6 +376,49 @@ define([
         });
     };
 
+    var supportsImageBitmapOptionsPromise;
+    /**
+     * A helper function to check whether createImageBitmap supports passing ImageBitmapOptions.
+     *
+     * @returns {Promise<Boolean>} A promise that resolves to true if this browser supports creating an ImageBitmap with options.
+     *
+     * @private
+     */
+    Resource.supportsImageBitmapOptions = function() {
+        // Until the HTML folks figure out what to do about this, we need to actually try loading an image to
+        // know if this browser supports passing options to the createImageBitmap function.
+        // https://github.com/whatwg/html/pull/4248
+        if (defined(supportsImageBitmapOptionsPromise)) {
+            return supportsImageBitmapOptionsPromise.promise;
+        }
+
+        supportsImageBitmapOptionsPromise = when.defer();
+
+        if (!FeatureDetection.supportsCreateImageBitmap()) {
+            supportsImageBitmapOptionsPromise.resolve(false);
+            return supportsImageBitmapOptionsPromise.promise;
+        }
+
+        var imageDataUri = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWP4////fwAJ+wP9CNHoHgAAAABJRU5ErkJggg==';
+
+        Resource.fetchBlob({
+            url : imageDataUri
+        })
+            .then(function(blob) {
+                return when(createImageBitmap(blob, {
```

Comment:
There's no need to wrap these calls in `when` if you are already inside of an existing when-based promise chain.

Reply:
`createImageBitmap` returns a native promise, not a `when` promise. I do this because catching errors for native promises is `.catch` instead of `.otherwise`.

Reply:
Yes, I understand why you are doing it, but it's not necessary because you are already inside of a when based promise.  Looks at the console output of this [Sandcastle demo](https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/#c=jZLLbtswEEV/ZaCNZCAgkW2iGq29CAIUDZCsCmgRihxZ04xIgY8obpF/L2W5tQtnUa5IDu+58+AWA6VBPGJwyWsUHUbdb9i11a/GwtlKnuEGyj7GMdxIqQ+6H0E4v5Pbw0FuErGRX8YxyCdljVYhMkqyBt9EHwcuT8T3VWNF7NFWXbI6krNVm01XcHTVzgbHKNjtqvLORZijV6AVM9kdaI8q4v2gdrihOKixXN0uQim/ObCIBqKDyasRWtQqBYQJQXkEsoFM3sLzlO2fYU7iwAXFmWr2YgF5jMlbmHL6bhIXhtWS0HmTaI4+eEIb1VxR7lbHNH4/lv0+p/hB3XSCflz+hTeElNsfyr9Al4l+ooAn6v+iOkWcPB5Zt8VVUYe4Z1wv6s80jM7HefqVEDLiMHImBNkm/YJR6BBmZS3/iGpDr0DmU1MsP2TrcjfIom8K0KxCyJEuMT/RT2yKdS3z+39k7JTJI354Rc9qPz/pr9dfl0shRC3z8VIVneNW+TPibw) showing what I mean.

Ultimately these wraps aren't needed, you only need to wrap it when it's a "top level" promise.